domain/src/main/kotlin/no/nav/su/se/bakover/domain/oppdrag/Utbetalingsstrategi.kt

```diff
@@ -61,40 +65,65 @@ sealed class Utbetalingsstrategi {
-                validate(Utbetaling.UtbetalingsType.STANS != it.type) { "Kan ikke stanse utbetalinger som allerede er stanset" }
-                validate(Utbetaling.UtbetalingsType.OPPHØR != it.type) { "Kan ikke stanse utbetalinger som allerede er opphørt" }
-            } ?: throw UtbetalingStrategyException("Ingen oversendte utbetalinger å stanse")
+        fun generer(): Either<Feil, Utbetaling.UtbetalingForSimulering> {
+            val sisteOversendteUtbetalingslinje =
+                sisteOversendteUtbetaling()?.sisteUtbetalingslinje() ?: return Feil.FantIngenUtbetalinger.left()
 
-            /**
-             * TODO jm: kan fjernes etter https://jira.adeo.no/browse/TOB-1772 er fikset.
-             */
-            unngåBugMedReaktiveringAvOpphørIOppdrag(
-                datoForStanEllerReaktivering = stansDato,
-            )
+            when {
+                !harOversendteUtbetalingerEtter(stansDato) -> {
+                    return Feil.IngenUtbetalingerEtterStansDato.left()
+                }
+                !stansDato.erFørsteDagIMåned() -> {
+                    return Feil.StansDatoErIkkeFørsteINesteMåned.left()
+                }
+                LocalDate.now(clock).plusMonths(1).startOfMonth() != stansDato.startOfMonth() -> {
+                    return Feil.StansDatoErIkkeFørsteINesteMåned.left()
+                }
+                sisteOversendteUtbetalingslinje is Utbetalingslinje.Endring.Stans -> {
+                    return Feil.SisteUtbetalingErEnStans.left()
+                }
+                sisteOversendteUtbetalingslinje is Utbetalingslinje.Endring.Opphør -> {
+                    return Feil.SisteUtbetalingErOpphør.left()
+                }
+                /**
+                 * TODO jm: kan fjernes etter https://jira.adeo.no/browse/TOB-1772 er fikset.
+                 */
+                unngåBugMedReaktiveringAvOpphørIOppdrag(stansDato).isLeft() -> {
+                    return Feil.KanIkkeStanseOpphørtePerioder.left()
+                }
+            }
 
             return Utbetaling.UtbetalingForSimulering(
+                opprettet = Tidspunkt.now(clock),
                 sakId = sakId,
                 saksnummer = saksnummer,
                 fnr = fnr,
                 utbetalingslinjer = nonEmptyListOf(
                     Utbetalingslinje.Endring.Stans(
-                        utbetalingslinje = sisteOversendtUtbetaling.sisteUtbetalingslinje(),
+                        utbetalingslinje = sisteOversendteUtbetalingslinje,
                         virkningstidspunkt = stansDato,
                         clock = clock,
                     ),
                 ),
                 type = Utbetaling.UtbetalingsType.STANS,
                 behandler = behandler,
                 avstemmingsnøkkel = Avstemmingsnøkkel(Tidspunkt.now(clock)),
-            )
+            ).right()
+        }
+
+        @Deprecated(message = "Erstattet for å kunne returnere Either", level = DeprecationLevel.ERROR)
```

Comment:
Fungerer bygget med denne? Kunne vi ikke heller fjernet den og gjort when der den blir brukt?

Reply:
Ja, så lenge metodene ikke er i bruk går det helt fint. Ble sånn mest i frykt for å måtte endre signaturer på de andre strategiene også, men når jeg tenker meg om trenger vi nok ikke å gjøre det uansett. Tror også den kan fjernes.